### ISSUE TITLE
Is there any difference in preprocess_function when train and inference? 

### ISSUE THREAD

**Issue by WatsonWangZh** [zjunlp/DeepKE#271]
**Describe the bug**
> A clear and concise description of what the bug is.

There is an error: ValueError: Column to remove ['output', 'kg'] not in the dataset. Current columns in the dataset: ['id', 'cate', 'instruction', 'input'] when do predict. and there is no need of label when do inference, right?

**Environment (please complete the following information):**
 - OS: [e.g. mac / window]
 - Python Version [e.g. 3.6]

**Screenshots**
> If applicable, add screenshots to help explain your problem.

**Additional context**
> Add any other context about the problem here.

**Comment by GoooDte** [comment]
Hello, please tell me where exactly did your bug appear?

**Comment by guihonghao** [comment]
I'm sorry, this is a minor error in the source code. We used "remove_columns" in datasets.map() to remove some columns that are not needed during training. However, "train.json" and "valid.json" have different columns. The "valid.json" does not include 'output' and 'kg'. Therefore, if you simultaneously perform "do_train" and "do_predict", an error will occur. We have made the necessary modifications.
